File: packages/folio-core/domain/worklog.go
package domain

import "time"

type TicketLog struct {
	ID        TicketLogID
	ProjectID ProjectID
	IssueID   IssueID
	CycleID   CycleID
	Body      string
	CreatedBy UserID
	CreatedAt time.Time
	UpdatedAt time.Time
}

type TicketLogFilter struct {
	IssueID IssueID
	CycleID  CycleID
	Search   string
	Limit    int
	Offset   int
}

type PlanLog struct {
	ID        PlanLogID
	ProjectID ProjectID
	PlanID    PlanID
	Body      string
	CreatedBy UserID
	CreatedAt time.Time
	UpdatedAt time.Time
}

type PlanLogFilter struct {
	PlanID PlanID
	Search string
	Limit  int
	Offset int
}

type TodoLog struct {
	ID        TodoLogID
	ProjectID ProjectID
	IssueID   IssueID
	Body      string
	CreatedBy UserID
	CreatedAt time.Time
	UpdatedAt time.Time
}

type TodoLogFilter struct {
	IssueID IssueID
	Search string
	Limit  int
	Offset int
}

File: apps/web/src/core/domain/worklog.ts
import type { Branded } from './branded'
import type { CycleId } from './cycle'
import type { PlanId } from './plan'
import type { ProjectId, UserId } from './project'
import type { TicketId } from './ticket'
import type { TodoId } from './todo'

export type WorkLogId = Branded<string, 'WorkLogId'>

export const workLogId = (value: string): WorkLogId => value as WorkLogId

type Base = {
	readonly id: WorkLogId
	readonly projectId: ProjectId
	readonly body: string
	readonly createdBy: UserId | undefined
	readonly createdAt: Date
	readonly updatedAt: Date
}

export type TicketLog = Base & {
	readonly ticketId: TicketId
	readonly cycleId: CycleId | undefined
}

export type PlanLog = Base & {
	readonly planId: PlanId
}

export type TodoLog = Base & {
	readonly todoId: TodoId
}
